Topics/Software_Tools/PyAutoGUI/Automation_with_PyAutoGUI.md

```diff
@@ -0,0 +1,140 @@
+# So the mouse moves to X of 100, Y of 500
+
+>>> pyautogui.moveTo(600, None)  # Similarly, moves mouse to X of 600, Y of 500.
+```
+
+`move()` on the other hand lets you move the mouse to a location relative to its current one.
+```python
+>>> pyautogui.move(10, 0) # move the mouse 10 pixels to the right
+>>> pyautogui.move(-50, 100) # move the mouse 50 pixels to the right and 100 pixels up
+```
+
+This movement is instantaneous, but both these functions also take a third argument, which can be used to change the duration of the movement in seconds.
+
+```python
+>>> pyautogui.move(10, 0, 2) # move the mouse 10 pixels to the right over 2 seconds
+>>> pyautogui.moveTo(100, 200, 10)   # moves mouse to X of 100, Y of 200 over 10 seconds
+```
+
+How do I know where to move my mouse? Well here is a simple trick that you can use to find the $(x, y)$ coordinates of a pixel on your screen:
+
+```python
+import pyautogui
+import time
+
+time.sleep(5) # 5 second buffer to let you move your mouse
+print(pyautogui.position()) # prints the current position of the mouse
+```
+All you have to do is simple run the program above and move your mouse to the location you want the coordinates for (make sure its within 5 seconds). The output will tell you exactly where the mouse is!
+
+### Clicking
+Now that you can move your mouse around freely to wherever you want, how do you finally perform a mouseclick? PyAutoGUI provides multiple options as follows:
+```python
+>>> pyautogui.click()  # perform a simple left-click
+>>> pyautogui.click(x=450, y=230)  # move to 450, 230, then left-click
+>>> pyautogui.click(button='right')  # right-click the mouse
+>>> pyautogui.doubleClick()  # double left-click
+>>> pyautogui.click(clicks=3)  # triple left-click
+>>> pyautogui.click(button='right', clicks=4, interval=0.5) #quadruple right-click 
+# but with a 0.5 second pause in between the clicks
+```
+
+### Other Cool Mouse Functions
+There's a lot more than just clicking that you can do with the PyAutoGUI mouse.
+Here are a few examples:
+```python
+>>> pyautogui.dragTo(300, 400, 2, button='left')  # drag mouse to X of 300, Y of 400 over 2 seconds while holding down left mouse button
+>>> pyautogui.drag(30, 0, 2, button='right')   # drag the mouse left 30 pixels over 2 seconds while holding down the right mouse button
+>>> pyautogui.scroll(10)   # scroll up 10 "clicks"
+>>> pyautogui.hscroll(-10)  # scroll left 10 "clicks"
+```
+Go check out the docs for a lot more!
+
+## Keyboard Controls
+Pending
+
+## Locating Images and Screenshots
+Let's say you want to click on something that's going to pop up on the screen but you don't know where it's going to be. PyAutoGUI has just the solution for that. 
+
+Suppose you want to open the UofT folder on your desktop below, but you don't want to bother finding its coordinates.
+![desktop](./assets/desktop.png)
+
+First take a screenshot of the folder and save it:
+
+![desktop](./assets/uoft_folder.png)
```

Comment:
Your inclusion of visuals detailing the process of capturing and saving screenshots greatly enhances the tutorial's clarity and usability. By providing clear instructions accompanied by images, readers can easily follow along and replicate the steps. This addition ensures a smoother learning experience, allowing CSC301 students to grasp the concept of image location effortlessly

Reply:
Thank you. I was thinking of including a master example in this section to round it off, showcasing the power of the library. Any thoughts?

Reply:
That's a great idea!